### PR TITLE
JTC trajectory end time validation fix

### PR DESCRIPTION
**Pull request by henrygerardmoore**
The `time_from_start` member for a `JointTrajectoryPoint` is monotonically increasing, not a delta time, so I think we can just use the last one in the vector; summing them up seems incorrect. I also moved the points empty check earlier because `.back().time_from_start` would seg fault with an empty points vector.

### CODE REVIEW

**Comment by christophfroehlich** [comment]
Could you please check the docs if this is clearly documented how to fill the `trajectory` field? [Here](https://control.ros.org/master/doc/ros2_controllers/joint_trajectory_controller/doc/trajectory.html#trajectory-replacement) is a comment, but maybe we should add this in a more prominent place.

**Comment by henrygerardmoore** [comment]
> Could you please check the docs if this is clearly documented how to fill the `trajectory` field? [Here](https://control.ros.org/master/doc/ros2_controllers/joint_trajectory_controller/doc/trajectory.html#trajectory-replacement) is a comment, but maybe we should add this in a more prominent place.

@christophfroehlich 
That page seems clear to me, the plots in there are very helpful. Perhaps it should be in the main JTC page, but where it is seems fine IMO.

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
LGTM. Just one request: Could you please add checks for the end time to `invalid_message` test?

File: joint_trajectory_controller/test/test_trajectory_controller.cpp
Comment (christophfroehlich) [inline joint_trajectory_controller/test/test_trajectory_controller.cpp:1391]: Is this enough to test? 
Maybe we should add two tests:
- One with start_time + time_from_start < now() -> fail
- One with start_time + time_from_start > now() -> success

Comment (henrygerardmoore) [reply joint_trajectory_controller/test/test_trajectory_controller.cpp:1391]: Good point, I have added one such successful case as well.

**Review by christophfroehlich — APPROVED** [review]
LGTM, thanks!

**Review by bijoua29 — APPROVED** [review]

**Comment by henrygerardmoore** [comment]
@christophfroehlich is there anything I can do to help get this across the finish line?

**Comment by christophfroehlich** [comment]
No, we have to wait for an approval of @destogl or @bmagyar

**Review by bmagyar — APPROVED** [review]

**Pull request merged**
